aten/src/ATen/native/AdaptiveMaxPooling2d.cpp

```diff
@@ -346,31 +395,83 @@ Tensor& adaptive_max_pool2d_backward_out_cpu_template(
 
 } // namespace
 
-std::tuple<Tensor&, Tensor&> adaptive_max_pool2d_out_cpu(const Tensor& input,
-  IntArrayRef output_size,
-  Tensor& output,
-  Tensor& indices)
-{
-  adaptive_max_pool2d_out_cpu_template(
-    output,
-    indices,
-    input,
-    output_size);
-  return std::tuple<Tensor&, Tensor&>(output, indices);
-}
+TORCH_IMPL_FUNC(adaptive_max_pool2d_out_cpu)
+(const Tensor& input, IntArrayRef output_size, const Tensor& output, const Tensor& indices) {
+  int dimW = 2;
+  int dimH = 1;
+  int64_t sizeB = 1;
+  int64_t sizeD = 0;
+  int64_t isizeH = 0;
+  int64_t isizeW = 0;
 
-std::tuple<Tensor, Tensor> adaptive_max_pool2d_cpu(
-  const Tensor& input,
-  IntArrayRef output_size)
-{
-  Tensor output = at::empty({0}, input.options());
-  Tensor indices = at::empty({0}, input.options().dtype(kLong));
-  adaptive_max_pool2d_out_cpu_template(
```

Comment:
@SplitInfinity should delete this function? the same as cuda.

Reply:
and should use `git rebase -i` to squash these two commits into one?

Reply:
I'll delete it. It has already been deleted in the CUDA version.

As for squashing commits, the merging automation will take care of that.